src/utils/string.cc

```diff
@@ -267,6 +267,28 @@ void replaceAll(std::string *str, const std::string& from,
     }
 }
 
+std::string log_escape_hex(std::string s) {
```

Comment:
The new functionality is highly comparable to the pre-existing function toHexIfNeeded().

The name for the new function proposed here doesn't really communicate what it's doing (the fn only cares about the double quote character) and it's unnecessarily dissimilar to the existing function.

I would suggest a name something like toHexSingleChar might be advantageous, along with an additional interface change:

It's pretty easy to make this new function at least slightly more generic by passing in the char we care about. I.e. the function can include a second parameter of type 'char' that indicates which character should be encoded when the function is called.

As for the guts of the function:

As stated earlier, the implementation details of this function are very similar to toHexIfNeeded(). It's preferable from a maintenance perspective not to adopt a different implementation methodology here. I.e. just copy the internals of the existing function and then change the 'if' condition.

Reply:
> The new functionality is highly comparable to the pre-existing function toHexIfNeeded().

yes, I saw that, but I didn't want to modify.

> The name for the new function proposed here doesn't really communicate what it's doing (the fn only cares about the double quote character) and it's unnecessarily dissimilar to the existing function.

the name comes from v2 naming - it's the same.
 
> I would suggest a name something like toHexSingleChar might be advantageous, along with an additional interface change:
> 
> It's pretty easy to make this new function at least slightly more generic by passing in the char we care about. I.e. the function can include a second parameter of type 'char' that indicates which character should be encoded when the function is called.
> 
> As for the guts of the function:
> 
> As stated earlier, the implementation details of this function are very similar to toHexIfNeeded(). It's preferable from a maintenance perspective not to adopt a different implementation methodology here. I.e. just copy the internals of the existing function and then change the 'if' condition.

Right, I see. I'm going to review and modify.

Reply:
Thanks for sharing your idea.

I modified the function, but a little differently than you described above. Hope you will like that.

> The name for the new function proposed here doesn't really communicate what it's doing (the fn only cares about the double quote character) and it's unnecessarily dissimilar to the existing function.

Not exactly. As I explained, I would have follow the behavior of v2 [log escape fn](https://github.com/SpiderLabs/ModSecurity/blob/v2/master/apache2/msc_util.c#L1304-L1307). That changes the characters below 31, above 127 or if it's the `"` or `\` (34 or 92).

The function [toHexIfNeeded()](https://github.com/SpiderLabs/ModSecurity/blob/v3/master/src/utils/string.cc#L139-L152) does the similar, except it does not escape `"` and `\`. Now I added a new prototype of the `toHexIfNeeded()` with an optional boolean argument, which marks the caller wants to encode special chars too (`"` and `\`). The old prototype is still exists, it calls the new one with `false` at the second argument.

Now it fits for the necessary behavior, and easy to understand it, I guess.

I keep this review unresolved, let me know if you have any additional note.